components/camel-ssh/src/main/java/org/apache/camel/component/ssh/SSHPublicKeyHolder.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.ssh;
+
+import java.math.BigInteger;
+import java.nio.charset.Charset;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.DSAPublicKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPublicKeySpec;
+
+public class SSHPublicKeyHolder {
+    private static final String SSH_RSA = "ssh-rsa";
+    private static final String SSH_DSS = "ssh-dss";
+    private static final String SSH_ECDSA = "ecdsa-sha2-nistp256";
+    private static final String SSH_ED25519 = "ssh-ed25519";
+
+    private String keyType;
+
+    /* RSA key parts */
+    private BigInteger e;

```

Comment:
Hi, the variable names are directly derived from the KeySpec format (which are derived from the RSA/DSA specification)